lib/report-generator.js

```diff
@@ -0,0 +1,93 @@
+var fs = require('fs-extra'),
+    path = require('path');
+
+/**
+ * The directory of the visual report
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var DIRECTORY_PATH = 'visual-report';
+
+/**
+ * The directory of the static files
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var STATIC_DIRECTORY = 'statics';
+
+/**
+ * The path of the data file
+ *
+ * @type {String}
+ * @const
+ * @default
+ */
+var DATA_PATH = 'data.js';
+
+/**
+ * Creates the files for the visual report
+ *
+ * @param {Array.<Suite>} data - The generated data from Mocha's runner
+ * @param {Object} options - Options object
+ * @param {String} [options.rootDirectory] - The directory of the visual report
+ * @param {generateReportCb} done - Cb to notify when the report is created
+ */
+function generateReport(data, options, done) {
+  if (data === undefined) {
+    throw new Error('No received data to create the "data.js" file');
+  }
+
+  if (done === undefined) {
+    throw new Error('No received callback, to notify when report is created');
+  }
+
+  var rootDirectory = options.rootDirectory || DIRECTORY_PATH;
+
+  try {
+    fs.mkdirSync(rootDirectory);
+  } catch(error) {
+    if (error.code !== 'EEXIST') {
+      return done(error);
+    }
+  }
+
+  var outputPath = path.join(rootDirectory, DATA_PATH);
+      output = 'var ' + generateReport._mochaTestData + ' = ' +
+        JSON.stringify(data, null, 2) + ';',
+
+  fs.writeFile(outputPath, output, function(error) {
+    if (error) {
+      return done(error);
+    }
+
+    var source = path.join(__dirname, '..', STATIC_DIRECTORY),
+        destination = path.join(rootDirectory, STATIC_DIRECTORY);
+
+    fs.copy(source, destination, function(error) {
+      if (error) {
+        return done(error);
+      }
+
+      done(null);
+    });
+  });
+}
+
+/**
+ * The var name from which the data will be retrieved in the browser
+ *
+ * @type {String}
+ */
+generateReport._mochaTestData = 'data';
```

Comment:
Don't make the var private. You actually want it public because whoever's using the function should know the name of the variable that gets written to that file. It should also be in CAPS because it's a constant. One last thing, `mochaTestData` gives the impression that you can find __data__ inside of it, when you actually find a name. So it should be named `MOCHA_TEST_DATA_VAR_NAME`.